mlflow/pipelines/pipeline.py

```diff
@@ -213,19 +212,14 @@ def get_artifact(self, artifact_name: str):
         Returns None if the specified artifact is not found.
         Raise an error if the artifact is not supported.
         """
-        pass
-
-    @experimental
-    @abc.abstractmethod
-    def _get_artifact_path(self, artifact_name: str):
-        """
-        Get a path of an artifact from pipeline output. artifact names can be obtained from
-        `Pipeline.inspect()` or `Pipeline.run()` output.
-
-        Returns None if the specified artifact is not found.
-        Raise an error if the artifact is not supported.
-        """
-        pass
+        for step in self._steps:
+            for artifact in step.get_artifacts():
+                if artifact.name == artifact_name:
+                    return artifact
```

Comment:
Can we do return artifact.load()

Reply:
@sunishsheth2009 i believe that if we return the artifact content here, then we would not be able to re-use this function to get artifact path via CLI.

Reply:
If folks prefer, we can rename this method to `_get_artifact` and then implement `get_artifact` as follows:
```
get_artifact(self, name):
    return self._get_artifact(name).load()
```
My 2c would be to do this only if this interface is preferable (quite possible!) rather than to avoid a breaking change. IMO it depends on whether users would ever want other information about an artifact, such as its path.

Reply:
Ah yes! That makes sense. But we could go back to similar pattern we had previously of `get_artifact` and `_get_artifact_path`  or something that where we have similar logic which could be abstracted out as well. 
But compromising the User experience for that wouldn't be ideal IMO.